presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClientModule.java

```diff
@@ -38,8 +39,8 @@ public void configure(Binder binder)
 
     @Provides
     @Singleton
-    public ConnectionFactory getConnectionFactory(BaseJdbcConfig config)
+    public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
     {
-        return new DriverConnectionFactory(new SQLServerDriver(), config);
+        return new DriverConnectionFactory(new SQLServerDriver(), config, credentialProvider);
```

Comment:
nit. Possible future refactor. You could bind `Driver.class` to `new SQLServerDriver()` instance, then you could simply inject that to `DriverConnectionFactory`. So you could remove method like these.